test/local/account_routes.js

```diff
@@ -356,3 +358,852 @@ test(
+          return P.resolve(device)
+        },
+        createSessionToken: function (st) {
+          sessionToken = st
+          sessionToken.data = crypto.randomBytes(32)
+          sessionToken.tokenId = crypto.randomBytes(32)
+          sessionToken.lastAuthAt = function () {}
+          return P.resolve(sessionToken)
+        },
+        emailRecord: function () {
+          return P.reject(error.unknownAccount(mockRequest.payload.email))
+        }
+      },
+      log: mockLog,
+      mailer: {
+        sendVerifyCode: function () {
+          return P.resolve()
+        }
+      }
+    })
+
+    return new P(function (resolve) {
+      getRoute(accountRoutes, '/account/create')
+        .handler(mockRequest, resolve)
+    })
+    .then(
+      function () {
+        t.equal(mockLog.activityEvent.callCount, 2)
+        t.equal(mockLog.activityEvent.args[1].length, 3)
+        t.equal(mockLog.activityEvent.args[1][0], 'device.created')
+        t.equal(mockLog.activityEvent.args[1][1], mockRequest)
+        t.deepEqual(mockLog.activityEvent.args[1][2], {
+          uid: uid.toString('hex'),
+          device_id: deviceId.toString('hex')
+        })
+      },
+      function () {
+        t.fail('request should have succeeded')
+      }
+    )
+  }
+)
+
+test(
+  '/account/login emits account.login activity event and saves metrics context',
+  function (t) {
+    var keyFetchToken, sessionToken, uid
+    var mockLog = mocks.mockLog({
+      activityEvent: sinon.spy()
+    })
+    var mockMetricsContext = {
+      save: sinon.spy()
+    }
+    var mockRequest = {
+      app: {
+        acceptLanguage: 'en',
+        clientAddress: '127.0.0.1'
+      },
+      headers: {
+        'user-agent': 'foo'
+      },
+      payload: {
+        email: crypto.randomBytes(16).toString('hex') + '@restmail.net',
+        authPW: crypto.randomBytes(32).toString('hex')
+      },
+      query: {
+        keys: 'true'
+      }
+    }
+    var accountRoutes = makeRoutes({
+      checkPassword: function () {
+        return P.resolve(true)
+      },
+      customs: {
+        check: function () {
+          return P.resolve()
+        }
+      },
+      db: {
+        createKeyFetchToken: function (kft) {
+          keyFetchToken = kft
+          keyFetchToken.data = crypto.randomBytes(32)
+          keyFetchToken.tokenId = crypto.randomBytes(32)
+          return P.resolve(keyFetchToken)
+        },
+        createSessionToken: function (st) {
+          sessionToken = st
+          sessionToken.data = crypto.randomBytes(32)
+          sessionToken.tokenId = crypto.randomBytes(32)
+          sessionToken.lastAuthAt = function () {}
+          return P.resolve(sessionToken)
+        },
+        emailRecord: function () {
+          uid = crypto.randomBytes(16)
+          return P.resolve({
+            authAt: Date.now(),
+            email: mockRequest.payload.email,
+            emailVerified: true,
+            uid: uid
+          })
+        }
```

Comment:
Noticed this too, for sign-in confirmation, I pulled them into shared [mock](https://github.com/mozilla/fxa-auth-server/blob/5bf5255b84d1c677b1a557b7dd19c8729ad5148f/test/mocks.js#L85). Maybe it would be worthwhile to pull this into a separate branch/issue to track?